ClinGen/clincoded
# 2188 - added new action review/effective date to UNC tracking message.

Changed the message "status" attribute from string to an object:
{
 "date": "2019-12-11T19:32:24.151Z",
 ...
 "status": "provisionally approved",
...
 }
To
{
 "date": "2019-12-11T19:32:24.151Z",
 ...
 "status": {
     "name": "provisionally approved",
     "date": "2019-10-01T12:10:55.051Z"
 }
...
 }

Steps to test:
If want to test locally with Kafka server, see how to set up in https://asconfluence.stanford.edu/confluence/display/CLINGEN/Set+up+local+kafka+server+for+testin

Dev instance: 2188-gc-unc-tracking.demo.clinicalgenome.org/

1. Log in and associate with an affiliation that can publish GDM (e.g. ClinGen Curator Test Affiliation 1)
2. Create a new Gene-Disease Record (GDM)
3. Add some evidences if you want to
4. Go to "Classification Matrix" page, save the classification
5. Enter an earlier date for "Date Received" and provisionally approve the classification
     5a. Check the debug console, see a log message "Post tracking data succeeded:..." with sent json message.  Open the json message and check the status object has the correct name and date (message.status.name, message.status.date)
     5b. If testing locally and have set up Kafka server locally (see above link), should see the json message appears in the consumer terminal
     5c. If there is error, should see log message "Post tracking data failed:..." and other "Error sending ..." message.  And an error message started with "** track-data: Error sending ..." is logged to apache error log
      Also test leaving "Date Received" blank, and set the status object date is same as date.  (message.status.date is same as message.date)
6. Test setting and not setting "Final Approval Date", and approve the classification.  Check message status as in step 5
7. Publish and unpublish classification, check json message to see message.status.date is same as message.date.
